x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtSignatureValidator.java

```diff
@@ -272,25 +286,28 @@ public void validate(String tokenPrincipal, SignedJWT signedJWT, ActionListener<
                 );
 
                 jwkSetLoader.reload(ActionListener.wrap(reloadResult -> {
-                    if (false == reloadResult.v1()) {
-                        // No change in JWKSet
-                        logger.debug("Reloaded same PKC JWKs, can't retry verify JWT token [{}]", tokenPrincipal);
+                    if (Arrays.equals(jwkSetLoader.getContentAndJwksAlgs().sha256(), initialJwksVersion)) {
```

Comment:
Yeah, you are right.  It would be possible for the algs and sha to mismatch...functionally there would be issues with competing definitions being applied in a extremely short time frame, but it is more correct to not split the reads. I will adjust as suggested.